internal/app/secretless/handlers/mysql/connection_details_test.go

```diff
@@ -0,0 +1,83 @@
+package mysql
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConnectionDetailsCorrectlyParsesConnectionDetails(t *testing.T) {
+	credentials := make(map[string][]byte)
+
+	credentials["host"] = []byte("myhost")
+	credentials["port"] = []byte("1234")
+	credentials["username"] = []byte("myusername")
+	credentials["password"] = []byte("mypassword")
+
+	expectedConnDetails := ConnectionDetails{
+		Host:     "myhost",
+		Port:     1234,
+		Username: "myusername",
+		Password: "mypassword",
+		Options:  make(map[string]string),
+	}
+
+	actualConnDetails, err := NewConnectionDetails(credentials)
+	assert.Nil(t, err)
+
+	assert.EqualValues(t, expectedConnDetails, *actualConnDetails)
```

Comment:
I could be wrong, but iirc this statement will error if the `err` is not nil.  Ie, it won't fail as an assertion, but will error with a go runtime error.  You might expect that it would stop processing if an assertion fails, but it doesn't work that way (again, iirc).  So you have to wrap the 2nd assertion in an `if`, sadly.

Reply:
Line 26 should take care of that sanity check